### DISCUSSION TITLE
Why multiple Locations per Thing?

### DISCUSSION THREAD

**Discussion by kjmitch** [FraunhoferIOSB/FROST-Server#1977] [General]
Does anyone know why the SensorThings specification allows/requires multiple `Location`s for a given `Thing`? The complication of having to get a specific entry from a collection of one `Location` for any `Thing` is currently getting in my way with the client library I'm using, but I'm also wondering why this arrangement makes sense logically.

I would think even a very large `Thing` with very disparate `Sensor`s could always be described and recorded as having a single current `Location`, or as being multiple `Thing`s if multiple `Location`s really made sense for it. Is there some design consideration I hadn't thought of?

**Comment by hylkevds** [comment]
This question would be better asked in the [SensorThings API GitHub project](https://github.com/opengeospatial/sensorthings/)...
The answer is [in the spec](https://docs.ogc.org/is/18-088/18-088.html#requirement-thing-relations):

> The Location entity locates the Thing. Multiple Things MAY be located at the same Location. A Thing MAY not have a Location. A Thing SHOULD have only one Location.
> However, in some complex use cases, a Thing MAY have more than one Location representations. In such case, the Thing MAY have more than one Locations. These Locations SHALL have different encodingTypes and the encodingTypes SHOULD be in different spaces (e.g., one encodingType in Geometrical space and one encodingType in Topological space).

We also have a demo where each Thing has 5 Locations that are all the same location, but in different resulutions/levels of details:
France: https://demography.k8s.ilt-dmz.iosb.fraunhofer.de/v1.1/Things(1957)/Locations
Demo Map: https://api4inspire.k8s.ilt-dmz.iosb.fraunhofer.de/servlet/is/163/

**Reply by kjmitch** [reply]
Thanks, I wasn't aware there was a repository for the API specification itself. The question not being a concern about this specific implementation is why I opened this discussion in General instead of Q&A.

So it seems the answer is that while a Thing can have only one actual location, the Location object is specifically a *representation* of a location, and there can be multiple representations of the same location. That makes perfect sense, thank you!
